jabgui/src/main/java/org/jabref/gui/util/DefaultDirectoryUpdateMonitor.java

```diff
@@ -0,0 +1,255 @@
+                            }
+                        }
+                    } else if (kind == StandardWatchEventKinds.ENTRY_DELETE) {
+                        // We only handle "ENTRY_DELETE" here, so the context is always a Path
+                        @SuppressWarnings("unchecked")
+                        WatchEvent<Path> ev = (WatchEvent<Path>) event;
+                        Path path = ((Path) key.watchable()).resolve(ev.context());
+                        if (FileUtil.getFileExtension(path).isEmpty()) {
+                            notifyAboutDirectoryDeletion(path);
+                            cleanupListenersOfADeletedDirectory(path);
+                        } else if (FileUtil.isPDFFile(path)) {
+                            notifyAboutPDFDeletion(path);
+                        }
+                    }
+                    key.reset();
+                }
+                Thread.yield();
+            }
+        } catch (IOException e) {
+            JabRefException exception = new WatchServiceUnavailableException(
+                    e.getMessage(), e.getLocalizedMessage(), e.getCause());
+            filesystemMonitorFailure.set(Optional.of(exception));
+            LOGGER.warn("Error during watching", e);
+        }
+    }
+
+    @Override
+    public boolean isActive() {
+        return filesystemMonitorFailure.get().isEmpty();
+    }
+
+    private void importFilesInDatabase(List<Path> files, BibDatabaseContext database) {
+        int nbPDFs = files.size();
+        if (nbPDFs > 0) {
+            ImportHandler importHandler = new ImportHandler(database, preferences, fileUpdateMonitor, this, undoManager, stateManager, dialogService, taskExecutor);
+            importHandler.importFilesInBackground(files, database, preferences.getFilePreferences(), TransferMode.LINK)
+                         .onSuccess(_ -> {
+                             if (nbPDFs == 1) {
+                                 dialogService.notify(Localization.lang("Successfully imported %0 PDF.", nbPDFs));
+                             } else {
+                                 dialogService.notify(Localization.lang("Successfully imported %0 PDFs.", nbPDFs));
+                             }
+                         }).onFailure(_ -> {
+                             if (nbPDFs == 1) {
+                                 dialogService.notify(Localization.lang("Failed to import %0 PDF.", nbPDFs));
+                             } else {
+                                 dialogService.notify(Localization.lang("Failed to import %0 PDFs.", nbPDFs));
+                             }
+                         })
+                         .executeWith(taskExecutor);
+        }
+    }
+
+    private void notifyAboutDirectoryCreation(Path newPath) {
+        Path parentPath = newPath.toAbsolutePath().getParent();
+        for (DirectoryUpdateListener listener : listeners.get(parentPath)) {
+            if (listener instanceof DirectoryGroup parentGroup) {
```

Comment:
Code like this should not be there, because the UpdateMonitor should be agnostic of the listeners -- the listener should handle this.

Also `importFilesInDatabase` should go to a listener.